nltk/tokenize/casual.py

```diff
@@ -156,7 +156,7 @@
 
     # Remaining word types:
     r"""
-    (?:[a-z][a-z'\-_]+[a-z])       # Words with apostrophes or dashes.
+    (?:\w[\w'\-_]+\w)              # Words with apostrophes or dashes.
```

Comment:
@nschneid Thank you. Do you think that, following your suggestion, this would be a better solution for the whole line?

    (?:[^\W\d_](?:[^\W\d_]|['\-_])+[^\W\d_])